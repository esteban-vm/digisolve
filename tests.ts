import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
