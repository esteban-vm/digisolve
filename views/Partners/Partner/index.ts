export { default /* type PartnerProps */ } from './Partner'
