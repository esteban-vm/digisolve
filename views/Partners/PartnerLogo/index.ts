export { default, type LogoProps } from './PartnerLogo'
