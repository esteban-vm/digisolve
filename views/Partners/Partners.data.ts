import type { PartnerProps } from './Partner'
import { uuid } from '@/helpers'

const text = 'Partner'
const width = 500
const height = 500

const partners: PartnerProps[] = [
  {
    id: uuid(),
    text,
    image: { src: '/img/brand-logo-1.png', width, height },
  },
  {
    id: uuid(),
    text,
    image: { src: '/img/brand-logo-2.png', width, height },
  },
  {
    id: uuid(),
    text,
    image: { src: '/img/brand-logo-3.png', width, height },
  },
  {
    id: uuid(),
    text,
    image: { src: '/img/brand-logo-4.png', width, height },
  },
  {
    id: uuid(),
    text,
    image: { src: '/img/brand-logo-5.png', width, height },
  },
  {
    id: uuid(),
    text,
    image: { src: '/img/brand-logo-6.png', width, height },
  },
  {
    id: uuid(),
    text,
    image: { src: '/img/brand-logo-7.png', width, height },
  },
  {
    id: uuid(),
    text,
    image: { src: '/img/brand-logo-8.png', width, height },
  },
]

export default partners
