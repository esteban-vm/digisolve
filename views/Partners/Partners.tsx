import type { FunctionalComponent } from '@/types'
import { LogoItem, type LogoProps } from '@/components'
import { brandLogos } from '@/utils/images'
import { styled } from '@/utils/styles'

export const logos = brandLogos.map((logo, index) => ({ logo, alt: `Brand Logo ${index + 1}` } as LogoProps))

const PartnersComponent: FunctionalComponent = (props) => {
  return (
    <section {...props}>
      <div className='row'>
        <h2 className='section_heading'>Featured Clients</h2>
        <h3 className='section_subheading'>
          We&apos;ve worked with some of the best companies in the world. Here are some of our Amazing Partners
        </h3>
      </div>
      <div className='row'>
        {logos.slice(0, 4).map((logo, index) => (
          <LogoItem key={index} {...logo} />
        ))}
      </div>
      <div className='row'>
        {logos.slice(4).map((logo, index) => (
          <LogoItem key={index} {...logo} />
        ))}
      </div>
    </section>
  )
}

const Partners = styled(PartnersComponent)`
  text-align: center;
`

export default Partners
