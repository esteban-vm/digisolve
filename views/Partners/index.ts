export { default as Partners, logos } from './Partners'
