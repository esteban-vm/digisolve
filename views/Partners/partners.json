[
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-1.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-2.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-3.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-4.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-5.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-6.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-7.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-8.png",
      "height": 500,
      "width": 500
    }
  }
]
