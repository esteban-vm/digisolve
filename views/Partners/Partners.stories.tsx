import type { Meta } from '@storybook/react'
import { styled } from '@/styles'
import Partners from './Partners'

const Wrapper = styled.div`
  width: 100%;
  background-color: var(--color-white);
`

export default {
  component: Partners,
  title: 'Views/Partners',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
} as Meta<typeof Partners>

export const PartnersView = () => <Partners />
