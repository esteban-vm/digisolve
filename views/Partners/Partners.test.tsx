import { render, cleanup, screen, create } from '@/tests'
import partners from './Partners.data'
import Partners from './Partners'

describe('ðŸ§ª PARTNERS:', () => {
  describe('display tests:', () => {
    /** Section title */
    const name = /^featured clients$/i

    beforeEach(() => {
      render(<Partners />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const section = screen.getByRole('region', { name })
      expect(section).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the title', () => {
        const title = screen.getByRole('heading', { name, level: 2 })
        expect(title).toBeInTheDocument()
        expect(title).toBeVisible()
      })

      it('the subtitle', () => {
        const subtitle = screen.getByRole('heading', { name: /^we've worked with/i, level: 3 })
        expect(subtitle).toBeInTheDocument()
        expect(subtitle).toBeVisible()
      })

      it('the partners', () => {
        const logoElements = screen.getAllByRole('article')
        expect(logoElements).toHaveLength(partners.length)

        for (const logo of logoElements) {
          expect(logo).toBeInTheDocument()
          expect(logo).toBeVisible()
        }
      })
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<Partners />)
      expect(tree).toMatchSnapshot()
    })
  })
})
