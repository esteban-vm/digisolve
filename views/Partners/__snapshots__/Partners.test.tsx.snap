// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª PARTNERS: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: center;
}

.emotion-1 img {
  width: 65%;
  height: auto;
}

<section
  aria-labelledby=":r0:"
  className="emotion-0"
>
  <div
    className="row "
    role="presentation"
  >
    <h2
      className="section__heading"
      id=":r0:"
    >
      Featured clients
    </h2>
    <h3
      className="section__subheading"
    >
      We've worked with some of the best companies in the world. Here are some of our Amazing Partners
    </h3>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby=":r1:"
        className="emotion-1"
      >
        <img
          alt="Partner"
          data-nimg="1"
          decoding="async"
          height={500}
          id=":r1:"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbrand-logo-1.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-1.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-1.png&w=1080&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={500}
        />
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby=":r2:"
        className="emotion-1"
      >
        <img
          alt="Partner"
          data-nimg="1"
          decoding="async"
          height={500}
          id=":r2:"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbrand-logo-2.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-2.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-2.png&w=1080&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={500}
        />
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby=":r3:"
        className="emotion-1"
      >
        <img
          alt="Partner"
          data-nimg="1"
          decoding="async"
          height={500}
          id=":r3:"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbrand-logo-3.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-3.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-3.png&w=1080&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={500}
        />
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby=":r4:"
        className="emotion-1"
      >
        <img
          alt="Partner"
          data-nimg="1"
          decoding="async"
          height={500}
          id=":r4:"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbrand-logo-4.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-4.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-4.png&w=1080&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={500}
        />
      </article>
    </div>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby=":r5:"
        className="emotion-1"
      >
        <img
          alt="Partner"
          data-nimg="1"
          decoding="async"
          height={500}
          id=":r5:"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbrand-logo-5.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-5.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-5.png&w=1080&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={500}
        />
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby=":r6:"
        className="emotion-1"
      >
        <img
          alt="Partner"
          data-nimg="1"
          decoding="async"
          height={500}
          id=":r6:"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbrand-logo-6.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-6.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-6.png&w=1080&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={500}
        />
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby=":r7:"
        className="emotion-1"
      >
        <img
          alt="Partner"
          data-nimg="1"
          decoding="async"
          height={500}
          id=":r7:"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbrand-logo-7.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-7.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-7.png&w=1080&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={500}
        />
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby=":r8:"
        className="emotion-1"
      >
        <img
          alt="Partner"
          data-nimg="1"
          decoding="async"
          height={500}
          id=":r8:"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbrand-logo-8.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-8.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-8.png&w=1080&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={500}
        />
      </article>
    </div>
  </div>
</section>
`;
